cmd/collector/app/handler/grpc_handler.go

```diff
@@ -104,7 +104,7 @@ func (c *batchConsumer) consume(ctx context.Context, batch *model.Batch) error {
 }
 
 func (c *batchConsumer) validateTenant(ctx context.Context) (string, error) {
```

Comment:
This line only became part of the PR because of the rename from `TenancyConfig` to `TenancyManager`. The collector doesn't use the interceptors, which were introduced for query.  This PR is large; I propose backporting the new query interceptor to the collector should be in a follow-up.

Reply:
+1